app.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  get,
+  param,
+  FindRoute,
+  ParseParams,
+  InvokeMethod,
+  Send,
+  Reject,
+  RestBindings,
+  SequenceHandler,
+  ParsedRequest,
+  ServerResponse,
+  RestComponent,
+  RestServer,
+} from '@loopback/rest';
+import {
+  LogComponent,
+  log,
+  LogBindings,
+  LogFn,
+  LogLevel,
+  LogLevelMixin,
+} from './src';
+import {inject} from '@loopback/context';
+import {Application} from '@loopback/core';
+
+const restSequenceActions = RestBindings.SequenceActions;
+
+class MyController {
+  @get('/')
+  @log(LogLevel.INFO)
+  hello() {
+    return 'Hi anonymous';
+  }
+
+  @get('/test')
+  @param.query.string('name')
+  @log(LogLevel.WARN)
+  helloName(name: string) {
+    return {name: name, msg: `Hi ${name}`};
+  }
+}
+
+class LogSequence implements SequenceHandler {
+  constructor(
+    @inject(restSequenceActions.FIND_ROUTE) protected findRoute: FindRoute,
+    @inject(restSequenceActions.PARSE_PARAMS)
+    protected parseParams: ParseParams,
+    @inject(restSequenceActions.INVOKE_METHOD) protected invoke: InvokeMethod,
+    @inject(restSequenceActions.SEND) protected send: Send,
+    @inject(restSequenceActions.REJECT) protected reject: Reject,
+    // We get the logger injected by the LogProvider here
+    @inject(LogBindings.LOG_ACTION) protected logger: LogFn,
+  ) {}
+
+  async handle(req: ParsedRequest, res: ServerResponse) {
+    // We define these variable outside so they can be accessed by logger.
+    let args: any = [];
+    let result: any;
+    let route: any;
+
+    try {
+      route = this.findRoute(req);
+      args = await this.parseParams(req, route);
+      result = await this.invoke(route, args);
+      this.send(res, result);
+    } catch (err) {
+      this.reject(res, req, err);
+    }
+
+    // We call the logger function given to us by LogProvider
+    this.logger(req.url, result);
```

Comment:
When the request is rejected with an error, this will log `undefined` as a result. Can we log the error instead please?